Source/WebCore/html/HTMLDetailsElement.cpp

```diff
@@ -138,20 +156,13 @@ void HTMLDetailsElement::attributeChanged(const QualifiedName& name, const AtomS
         if (oldValue != m_isOpen) {
             RefPtr root = shadowRoot();
             ASSERT(root);
-            if (m_isOpen)
+            if (m_isOpen) {
                 root->appendChild(*m_defaultSlot);
-            else
+                queueDetailsToggleEventTask(DetailsState::Open, DetailsState::Closed);
```

Comment:
This looks backwards, oldState (first param) should be closed, and newState should be open